src/App.tsx

```diff
@@ -1,14 +1,56 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState('all');
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [query, setQuery] = useState('');
+
+  const clearSelectedTodo = () => {
+    setTodo(null);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(result => {
+        setTodos(result);
+        setIsLoaded(true);
+      });
+  }, []);
+
+  const lowerText = (str: string) => (
+    str.toLowerCase().includes(query.toLowerCase())
+  );
+
+  const filterTodos = todos
+    .filter(todoItem => {
+      switch (filterBy) {
+        case 'active':
+          return !todoItem.completed;
+
+        case 'completed':
+          return todoItem.completed;
+
+        case 'all':
+        default:
+          return todoItem;
```

Comment:
Create an enum for filter types and use it everywhere instead of `active`, `completed`, and `all`